agents/redfish/fence_redfish.py

```diff
@@ -0,0 +1,152 @@
+#!@PYTHON@ -tt
+
+# Copyright (c) 2018 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# Fence agent for devices that support the Redfish API Specification.
+
+import sys
+import re
+import json
+import requests
+import atexit
+sys.path.append("@FENCEAGENTSLIBDIR@")
+
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+from fencing import *
+from fencing import fail_usage, run_delay
+
+def get_power_status(conn, options):
+    uri = options["--systems-uri"]
+    response = send_get_request(options, uri)
+    if response['ret'] is False:
+        fail_usage("Couldn't get power information")
+    data = response['data']
+    if data[u'PowerState'].strip() == "Off":
+        return "off"
+    else:
+        return "on"
+
+def set_power_status(conn, options):
+    action = {
+        'on' : "On",
+        'off': "ForceOff",
+        'reboot': "GracefulRestart"
```

Comment:
Is there a hard-reset version we can use instead of GracefulRestart?

GracefulRestart might get stuck on e.g. I/O or systemd/init-scripts stopping processes, which will make the boot process take a long time or "forever" in worst-case scenario.

Reply:
Yes there is a ForceRestart option. I will add it.